filament/backend/src/metal/MetalContext.mm

```diff
@@ -153,5 +153,61 @@ bool isInRenderPass(MetalContext* context) {
     return context->currentRenderPassEncoder != nil;
 }
 
+void MetalPushConstantBuffer::setPushConstant(PushConstantVariant value, uint8_t index) {
+    if (mPushConstants.size() <= index) {
+        mPushConstants.resize(index + 1);
+        mDirty = true;
+    }
+    if (UTILS_LIKELY(mPushConstants[index] != value)) {
+        mDirty = true;
+        mPushConstants[index] = value;
+    }
+}
+
+void MetalPushConstantBuffer::setBytesIfDirty(
+        id<MTLCommandEncoder> encoder, ShaderStage stage) {
+    if (!mDirty) {
+        return;
+    }
+    constexpr size_t PUSH_CONSTANT_SIZE_BYTES = 4;
+    constexpr size_t PUSH_CONSTANT_BUFFER_INDEX = 26;
+
+    size_t bufferSize = PUSH_CONSTANT_SIZE_BYTES * mPushConstants.size();
+    char* buffer = static_cast<char*>(malloc(bufferSize));
```

Comment:
can this be  a scratch buffer pre-allocated the driver/global space?

Reply:
Yeah, we can probably allocate a static buffer and only use malloc if we need more space.

Reply:
note that the max size of number of push constant if 32 (or 128 bytes) due to vk spec limit.

Reply:
Ah okay that's good to know! In that case I can definitely allocate static storage space. I also just saw the `MAX_PUSH_CONSTANT_COUNT` that you added, so I'll use that.